### ISSUE TITLE
consecutive comparison operators

### ISSUE THREAD

**Issue by vprevosto** [acsl-language/acsl#16]
chains of `!=` are explicitely forbidden. Would it make sense to consider that `a != b != c` means `a!=b && a!=c && b!=c` in the same spirit as the `distinct` operator of SMTLib?

Of course, `!=` could not be mixed with any other operator.

**Comment by davidcok** [comment]
This would be a different chain than for other consecutive operators, which would mean a!= b && b != c.
As consecutive operators are only used for predicates and already cause ambiguity problems, I'm not eager to expand their use. I'd prefer an explicit \distinct within ACSL.

**Comment by vprevosto** [comment]
Fair enough. `\distinct` has some uses, so I'm leaving the issue opened but will edit it in that direction
